src/sql/workbench/browser/designer/tableActions.ts

```diff
@@ -16,13 +16,53 @@ export interface DesignerTableActionContext {
 	selectedRow?: number;
 }
 
-export class AddRowAction extends Action {
-	public static ID = 'designer.addColumnAction';
+export class DesignerTableAction extends Action {
+
+	private _table: Table<Slick.SlickData>;
+
+	constructor(
+		id: string,
+		label: string,
+		icon: string,
+		private needsRowSelection: boolean
+	) {
+		super(id, label, icon);
+	}
+
+
+	public set table(table: Table<Slick.SlickData>) {
+		this._table = table;
+	}
+
+	public updateState(row?: number) {
```

Comment:
the base class shouldn't need to know the sub classes, please override this method in MoveRowDownAction and MoveRowUpAction.